diff --git a/dist/index.js b/dist/index.js
--- a/dist/index.js
+++ b/dist/index.js
@@ -23,7 +23,7 @@
  * for a particular string, we need to force that string into the right locale.
  * @see https://github.com/leinelissen/embedded-postgres/issues/15
  */
-const LC_MESSAGES_LOCALE = 'en_US.UTF-8';
+const LC_MESSAGES_LOCALE = 'C';
 // The default configuration options for the class
 const defaults = {
     databaseDir: path.join(process.cwd(), 'data', 'db'),
@@ -43,8 +43,11 @@
 const BIN_PERMISSIONS = 0b101101101;
 const ensureBinIsExecutable = (filePath) => __awaiter(void 0, void 0, void 0, function* () {
+    if (platform() === 'win32') {
+        return;
+    }
     // Only fix the file's mode if it's missing a permission. This is useful
     // when the binaries are in a read-only file system, as a call to chmod
     // (even unnecessary) would cause a crash.
     const stat = yield fs.stat(filePath);
     if ((stat.mode & BIN_PERMISSIONS) !== BIN_PERMISSIONS) {
         yield fs.chmod(filePath, stat.mode | BIN_PERMISSIONS);
@@ -121,9 +124,9 @@
             const randomId = crypto.randomBytes(6).readUIntLE(0, 6).toString(36);
             const passwordFile = path.resolve(tmpdir(), `pg-password-${randomId}`);
             yield fs.writeFile(passwordFile, this.options.password + '\n');
             // Make the files executable, in case they are not
-            ensureBinIsExecutable(postgres);
-            ensureBinIsExecutable(initdb);
+            yield ensureBinIsExecutable(postgres);
+            yield ensureBinIsExecutable(initdb);
             // Initialize the database
             yield new Promise((resolve, reject) => {
                 var _a;
@@ -133,7 +136,7 @@
                     `--pwfile=${passwordFile}`,
                     `--lc-messages=${LC_MESSAGES_LOCALE}`,
                     ...this.options.initdbFlags,
-                ], Object.assign(Object.assign({}, permissionIds), { env: { LC_MESSAGES: LC_MESSAGES_LOCALE } }));
+                ], Object.assign(Object.assign({}, permissionIds), { env: Object.assign(Object.assign({}, globalThis.process.env), { LC_MESSAGES: LC_MESSAGES_LOCALE }) }));
                 // Connect to stderr, as that is where the messages get sent
                 (_a = process.stdout) === null || _a === void 0 ? void 0 : _a.on('data', (chunk) => {
                     // Parse the data as a string and log it
@@ -168,7 +171,7 @@
                 throw new Error('Postgres cannot run as a root user. embedded-postgres could not find a postgres user to run as instead. Consider using the `createPostgresUser` option.');
             });
             // Make the file executable, in case it is not
-            ensureBinIsExecutable(postgres);
+            yield ensureBinIsExecutable(postgres);
             yield new Promise((resolve, reject) => {
                 var _a;
                 // Spawn a postgres server
@@ -177,7 +180,7 @@
                     '-p',
                     this.options.port.toString(),
                     ...this.options.postgresFlags,
-                ], Object.assign(Object.assign({}, permissionIds), { env: { LC_MESSAGES: LC_MESSAGES_LOCALE } }));
+                ], Object.assign(Object.assign({}, permissionIds), { env: Object.assign(Object.assign({}, globalThis.process.env), { LC_MESSAGES: LC_MESSAGES_LOCALE }) }));
                 // Connect to stderr, as that is where the messages get sent
                 (_a = this.process.stderr) === null || _a === void 0 ? void 0 : _a.on('data', (chunk) => {
                     // Parse the data as a string and log it
